Extension/src/platform.ts

```diff
@@ -47,10 +47,8 @@ export class PlatformInformation {
                 throw new Error(localize("unknown.os.platform", "Unknown OS platform"));
         }
 
-        return Promise.all([distributionPromise, versionPromise])
-            .then(([distro, version]) =>
-                new PlatformInformation(platform, architecture, distro, version)
-            );
+        const [distro, version] = await Promise.all([distributionPromise, versionPromise]);
```

Comment:
I don't think this `Promise.all` adds any value.  We should just `await` the calls that matter in the switch..case statement and assign the variables directly there.

Then you just initialize `distro` and `version` as undefined instead of having `distributionPromise` and `versionPromise`

Reply:
please take a look at the modified version.